CHANGELOG.md

```diff
@@ -2,12 +2,56 @@
 
 ### Added
-- Introduce new sub-command `solang idl` which can be used for generating
+- **Solana / breaking:** The try-catch construct is no longer permitted on Solana, as it
+   never worked. Any CPI error will abort the transaction.
+   [seanyoung](https://github.com/seanyoung)
+- **Solana:** Introduce new sub-command `solang idl` which can be used for generating
   a Solidity interface file from an Anchor IDL file. This can be used for calling
   Anchor Contracts on Solana. [seanyoung](https://github.com/seanyoung)
+- **Substrate:** Provide specific Substrate builtins via a "substrate" file. The
+  `Hash` type from `ink!` is the first type made available for Solidity contracts.
+  [xermicus](https://github.com/xermicus)
+- **Substrate:** Introduce the `--log-api-return-codes` CLI flag, which changes the
+  emitted code to print return codes for `seal` API calls into the debug buffer.
+  [xermicus](https://github.com/xermicus)
+- Introduce function name mangling for overloaded functions and constructors, so
+  that they can be represented properly in the metadata.
+  [xermicus](https://github.com/xermicus)
+
+### Changed
+ - The Solana target now uses Borsh encoding rather than eth abi
+   encoding. This is aimed at making Solang contracts Anchor compatible.
+   [LucasSte](https://github.com/LucasSte)
+- **Substrate / breaking:** Supported node version is now pallet contracts `v0.22.1`.
+  [xermicus](https://github.com/xermicus)
+- **Substrate / breaking:** Remove the deprecated `random` builtin.
+  [xermicus](https://github.com/xermicus)
+
+### Fixed
+ - The parser does not give up after the first error, when possible.
+   [salaheldinsoliman](https://github.com/salaheldinsoliman)
+ - Constant expressions are checked for overflow.
+   [salaheldinsoliman](https://github.com/salaheldinsoliman)
+ - AddMod and MulMod were broken. This is now fixed.
+   [LucasSte](https://github.com/LucasSte)
+- **Substrate / breaking:** Solang is now compatible with `ink!` version 4 (beta).
+  [xermicus](https://github.com/xermicus)
+- **Substrate:** Switched ABI generation to use official `ink!` crates, which fixes all
+  remaining metadata regressions.
+  [extraymond](https://github.com/extraymond) and [xermicus](https://github.com/xermicus)
+- **Substrate:** Allow constructors to have a name, so that multiple constructors are 
+  supported, like in `ink!`.
+  [xermicus](https://github.com/xermicus)
+- Provided examples and most of the solidity code snippets in our documentation
+  are now checked for succesful compilation on the Solang CI.
+  [xermicus](https://github.com/xermicus)
```

Comment:
```suggestion
- Provided examples and checked most of the solidity code snippets in our documentation
  for succesful compilation on the Solang CI.
  [xermicus](https://github.com/xermicus)
```
There needs to be parallelism between the verbal usage after and before the `and`.

Reply:
Sorry I don't get this one. It says that both, the "provided examples" and "most of the solidity code snippets in our documentation" are check for succesfull compilation? What is wrong with this sentence?

Reply:
@LucasSte I rephrased that sentence, WDYT?

Reply:
> Sorry I don't get this one. It says that both, the "provided examples" and "most of the solidity code snippets in our documentation" are checked for successful compilation? What is wrong with this sentence?

Before the `and` you wrote `provided` without a subject. After the `and` you wrote `are checked` (in passive voice) with a subject. Whatever construction you choose, it must remain unchanged before and after the `and`. If you choose to use active voice and a subject, this construction must be the same for both clauses (before and after the preposition).

Reply:
> @LucasSte I rephrased that sentence, WDYT?

Sounds better now.